frontend/admin/src/js/components/apiManagedTable/basicTableHelpers.tsx

```diff
@@ -1,16 +1,93 @@
 import { InputMaybe, OrderByClause, SortOrder } from "@common/api/generated";
 import React from "react";
+import { IntlShape } from "react-intl";
+import { IndeterminateCheckbox } from "../Table/tableComponents";
 
 export type ColumnsOf<T extends Record<string, unknown>> = Array<Column<T>>;
 
 // Information about a column in a table
 export type Column<T> = {
   label: string;
+  header?: React.ReactNode;
   id: IdType<T>;
   accessor: (object: T) => React.ReactNode;
   sortColumnName?: string; // API requires exact DB column name for sorting, not the name of the field in the API 😢
 };
 
+// This defines the row selection column
+export function rowSelectionColumn<T>(
+  intl: IntlShape,
+  selectedRows: T[],
+  pageSize: number,
+  onRowSelectionChange: (e: RowSelectedEvent<T>) => void,
+): Column<T> {
+  return {
+    label: intl.formatMessage({
+      defaultMessage: "Selection",
+      description:
+        "Label for the row-selection column in the tables column-selection modal.",
+    }),
+    header: (
+      <IndeterminateCheckbox
+        checked={selectedRows.length === pageSize}
+        indeterminate={
+          selectedRows.length > 0 && selectedRows.length < pageSize
+        }
+        onChange={() => {
+          onRowSelectionChange({ setSelected: selectedRows.length < pageSize });
+        }}
+      />
+    ),
+    accessor: (r) => {
+      return (
+        <input
+          type="checkbox"
+          checked={selectedRows.includes(r)}
+          onChange={() => {
+            onRowSelectionChange({
+              row: r,
+              setSelected: !selectedRows.includes(r),
+            });
+          }}
+        />
```

Comment:
Can we add an `id` to this checkbox so that we can label it? I'm not exactly sure the best way to do this though....hmm.

Possibly this?

```tsx
// basicTableHelpers.tsx
<input id={`select-${r.id}`} ...  />

// UserTable.tsx
{
  ...
  accessor: (user) => <label htmlFor={`select-${user.id}`}>{fullName(user)}</label>,
  id: "candidateName",
  ...
},
```

Reply:
Oh, I didn't see your message. I think that's the same as what I just mentioned though.

Reply:
@esizer You know more about accessibility than I do. Do you have anything to add?

Reply:
I think there are three solutions, none of them are great honestly but the whole discussion around this pattern are shaky at best anyway.

> Ordered based on effort.

1. **Easy**: Combine the columns
    - This works for hit state + label
    - Not ideal because we diverge from the prototype 
2. **Medium**: Separate columns and add custom checkbox to increase hit state
    - Kind of odd and not necessarily best practice
    - Could take a bit of time
 3. **Hard**: Replace the checkboxes with buttons
    - More control over styling (size)
    - We can simply add hidden text in the button as a label
    - Lot's of rework for James

Reply:
the Unselect part seems to be buggy. But otherwise I think it works ok. To be honest I'm not super happy with our solution, but I have reached out to some A11y peepz in the hopes of getting a better approach. My guess is that this will eventually be a very used feature, but I would be happy with creating a ticket to implement a better solution later. As long as we have something decent right now for this sprint.

(Bug: it's not unselecting all, it should alternate between selecting all and unselecting all)
![chrome_Dh6Gmc46L1](https://user-images.githubusercontent.com/16977254/177910499-b406d81e-1fa9-45fd-aa49-f23545b26d88.gif)